wsd/FileServer.cpp

```diff
@@ -1069,6 +1074,63 @@ void FileServerRequestHandler::preprocessFile(const HTTPRequest& request,
     Poco::replaceInPlace(preprocess, std::string("%FEEDBACK_URL%"), std::string(FEEDBACK_URL));
     Poco::replaceInPlace(preprocess, std::string("%WELCOME_URL%"), std::string(WELCOME_URL));
     Poco::replaceInPlace(preprocess, std::string("%BUYPRODUCT_URL%"), buyProduct);
+    Poco::URI indirectionURI(config.getString("indirection_endpoint.url", ""));
+    bool validIndirection = !indirectionURI.empty() && !wopiSrc.empty();
+    std::string routeToken = std::string(), serverId = std::string();
+#if !ENABLE_DEBUG
+    if (validIndirection && Util::iequal(indirectionURI.getScheme(), "http"))
+    {
+        LOG_ERR("Indirection url should only use HTTPS protocol: " << indirectionURI.toString());
+        validIndirection = false;
+    }
+#endif
+    if (validIndirection)
+    {
+        try
+        {
+            indirectionURI.addQueryParameter("WOPISrc", wopiSrc);
+            auto httpSession = http::Session::create(indirectionURI.getHost(),
+                                                     http::Session::Protocol::HttpSsl,
+                                                     indirectionURI.getPort());
+
+            http::Request operatorRequest(indirectionURI.getPathAndQuery());
+
+            const std::shared_ptr<const http::Response> httpResponse =
```

Comment:
So ... I know this will make things quite a bit more complicated. But we cannot have a synchronous request inside this.
That will effectively stall all web-page serving until this requests completes - which will very significantly increase the latency for anyone else connecting to this server.

We should instead complete the web request in a closure of an async fetch of this information. Probably we should do this very early / before this pre-processing method starts. Then we can pass the fetched information into the method which will pre-process and return.

That lets us get back to the main-loop in the meantime and serve lots of other requests - there are quite a lot on first login eg. Thanks !